EIPS/eip-6228.md

```diff
@@ -0,0 +1,233 @@
+---
+eip: 6228
+title: Extreme ЕRС-20
+description: meta-transaction token (MTT)
+author: Sam Porter (@SamPorter1984)
+discussions-to: https://ethereum-magicians.org/t/eip-6228-extreme-erc-20-meta-transaction-token-mtt/12300
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-12-26
+requires: 20
+---
+
+## Abstract
+
+This standard proposes a solution in which meta-transaction fees are deducted directly from the user' token transfer amount, no liquidity for keeper incentives required.
+
+This EIP is based on these assumptions:
+
+- That meta-transaction signatures need to be public in order to speed up blockchain adoption. The EIP assumes that token dApps will feature a public pool of signatures similar to Coinbase or Etherscan, which will resemble regular Ethereum transactions but with reduced user friction.
+- Meta-transactions must become standard procedure when interacting with smart contracts.
+- Gas efficiency is in no way less important when it comes to optimistic and zero knowledge roll-ups. If anything, lower network fees enable more automation, which is about to become essential in such a hyper-competitive industry.
+- Full automation can easily be too costly for project funding. Therefore, meta-transactions fees should be deducted directly from the transfer amount.
+
+An example:
+Alice has never had a wallet before but is eligible for an airdrop on Arbitrum. All she needs to do is install a wallet, sign the data, and then she has airdrop tokens minus the keeper fee. She does not have to make any transactions or possess any ETH. She is not limited in any way more than Bob, who is transacting on his own. Alice can vote, stake, claim rewards, trade on Uniswap. In fact, she can do more than Bob, such as subscribing to staking rewards to get them automatically every month, a part of these rewards then automatically sent to pay her bills and support her favorite content creators.
+
+## Motivation
+
+To solve the issues with existing keeper/relayer solutions today:
+
+- Conservative approaches massively limit the functionality, resulting in a lack of significant impact and usage. This EIP tries to acknowledge and embrace the “wild west” of DeFi without introducing more rules than are absolutely necessary.
+- All existing solutions rely on funding/supporting a relayer/keeper network, which makes innovation more dependent on external funding.
+
+## Specification
+
+Signatories can create executable signatures for transactions of any kind. As long as signatory `address(this)` token balance or `returnToken`(token returned after after execution) balance is higher than keeper fee of a transaction at hand, and as long as remaining value after keeper fee transaction is above `minOut`, it is executable.
```

Comment:
While I'm generally ok with jumping straight into technical content, reading EIP-20 and the text of this proposal up to this point doesn't give enough background for me to understand all the terms you're using. Are these terms/patterns common in the meta-transaction world? I'd really like to see some definitions and background information included if possible.